engine/src/stack.h

```diff
@@ -300,6 +300,9 @@ class MCStack : public MCObject, public MCMixinObjectHandle<MCStack>
     
     // MW-2014-09-30: [[ ScriptOnlyStack ]] If true, the stack is a script-only-stack.
     bool m_is_script_only : 1;
+    
+    // BWM-2017-08-16: [[ Bug 17810 ]] Line endings for imported script-only-stack.
+    MCStringLineEndingStyle m_line_encoding_style : 2;
```

Comment:
So, on the bitfield thing - I had forgotten that particular 'discussion' ;)

Bitfields are more than portable as long as you use them with care and don't use them in structs which are directly serialized... However *care* is needed.

The reason I suggested making this `: 2` was that it is between two other bitfields, so for consistency it seemed to make sense. However, looking at the entire MCStack header then it looks like it could save maybe 32 bytes or so by re-arrangement of the fields, but hardly essential (given the overhead of a stack in general, is quite large!).

It would have been fine if I had remembered that the signedness of bitfields vary - so this should be `: 3` to ensure that the top-bit is free (rule is always use 1 bit more than the number of bits needed to encode an enum's maximum value - then you don't get bit by the sign problem).

We perhaps need to update the style guide with better rules about field layout in classes and maybe still allow bitfields as long as they follow appropriate rules. Using a bitfield is pointless if there are less than 8 (8-bit or less sized) fields in a single struct/class, if there are many though they can make layouts more memory efficient; on the other hand, you can't default init them (i.e. bool m_is_script_only = false -- works; bool m_is_script_only : 1 = false -- doesn't) which means you have to have a constructor in a class which has bitfields. Swings, roundabouts!